dill/nonlocals.py

```diff
@@ -0,0 +1,122 @@
+# This file is only needed for PyPy2. It can be removed when support is dropped.
+# https://code.activestate.com/recipes/578965-python-2-nonlocal/
+
+import inspect, types, dis
+
+__all__ = ['export_nonlocals', 'nonlocals']
+
+# http://www.jonathon-vogel.com/posts/patching_function_bytecode_with_python/
+def find_code(code, f):
+    i = 0
+    while i < len(code):
+        if f(code, i):
+            return i
+        elif code[i] < dis.HAVE_ARGUMENT:
+            i += 1
+        else:
+            i += 3
+
+# http://nedbatchelder.com/blog/201301/byterun_and_making_cells.html
+def make_cell(value):
```

Comment:
This is basically the same as `CellType` (with `value=0`)...   It looks like you just cut-pasted from the URL in the header.  Can you check if there's some sort of copyright and/or copying instructions?

Reply:
Yep. It is the same. I had to take his copy as well because of circular imports. I think I could get away with it if I took the function directly from `sys.modules` instead of importing, but it seemed a little messy. I'll look to see if he has a license agreement. If so, I'll just save the resulting bytecode and create the code object directly.

Reply:
Just to be clear, I meant check for agreements for using the entire file.

Reply:
According to the page, the file uses the MIT license. It should probably be OK if we copy the entire file and put the license at the top.

Reply:
LGTM